### DISCUSSION TITLE
How to validate value/type of directive argument while schema validation?

### DISCUSSION THREAD

**Discussion by LastDragon-ru** [webonyx/graphql-php#1539] [Q&A] [answered]
I want to detect if the value of the argument has a valid type or not, eg in

```graphql
directive @a(a: String!) on FIELD_DEFINITION

type Query {
    a: String @a(a: {a: 123})
}
```

the expected error should be something like "`@a(a)` must be `String`". 

But the schema passes the validation  😱

```php
BuildSchema::build($schema)->assertValid(); // no error :(
```

Am I missing something?

**Comment by spawnia** [comment, marked as answer]
I believe validation for this is simply not implemented as of now. The reference implementation graphql-js also seems to lack it - see https://github.com/graphql/graphql-js/issues/1389. Since we base this library off of them, it just has not been added. Open for pull requests though!

**Reply by LastDragon-ru** [reply]
Ok, thanks.
